libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -269,7 +269,7 @@ class MapboxNavigation @VisibleForTesting internal constructor(
                 isFallback = false,
                 tilesVersion = navigationOptions.routingTilesOptions.tilesVersion
             ),
-            historyRecorder.fileDirectory(),
+            historyRecorder.historyRecorderHandle,
```

Comment:
Noting that the `nativeRouter` is `lazy` and the `Navigator` and the `HistoryRecorderHandle` are already created by the time the `nativeRouter` is referenced. First thing done in `MapboxNavigation` `init` 👀 https://github.com/mapbox/mapbox-navigation-android/blob/71db59a8aa054c1dda0ade04c70fbe08cbfaaad9/libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt#L412-L427